scripts/training/finetune-ldm-palette-adapter.py

```diff
@@ -0,0 +1,412 @@
+import random
+from functools import cached_property
+from typing import Any, TypedDict
+
+import datasets
+from loguru import logger
+from PIL import Image
+from pydantic import BaseModel
+from torch import Tensor, device as Device, dtype as DType
+from torch.distributions import Beta
+from torch.nn.functional import mse_loss
+from torch.utils.data import Dataset
+from torchvision.transforms import Compose, RandomCrop, RandomHorizontalFlip
+
+import refiners.fluxion.layers as fl
+from refiners.fluxion.utils import save_to_safetensors
+from refiners.foundationals.clip.text_encoder import CLIPTextEncoderL
+from refiners.foundationals.latent_diffusion.palette_adapter import ColorEncoder, PaletteAdapter
+from refiners.foundationals.latent_diffusion.schedulers.ddpm import DDPM
+from refiners.foundationals.latent_diffusion.stable_diffusion_1.model import SD1Autoencoder
+from refiners.foundationals.latent_diffusion.stable_diffusion_1.unet import SD1UNet
+from refiners.training_utils.callback import Callback
+from refiners.training_utils.config import BaseConfig
+from refiners.training_utils.latent_diffusion import (
+    LatentDiffusionConfig,
+    TestDiffusionConfig,
+    resize_image,
+    sample_noise,
+)
+from refiners.training_utils.trainer import Trainer
+
+Image.MAX_IMAGE_PIXELS = None
```

Comment:
Adding a comment would not hurt (about why you turn off this "decompression bombs" safety feature - is it really needed?)

Reply:
I'll add a comment, the default limit for [MAX_IMAGE_PIXELS](https://pillow.readthedocs.io/en/stable/reference/Image.html#PIL.Image.MAX_IMAGE_PIXELS) (89_478_485 pixels) is often lower than some of the images of the unsplash-lite dataset. For example [this image](https://images.unsplash.com/photo-1572371733185-448e1d92ea56) is 16384*10923=178_962_432 pixels.

Reply:
I'll set `MAX_IMAGE_PIXELS` to 200_000_000:
```python
>>> ds = datasets.load_dataset("1aurent/unsplash-lite", split="train")
>>> ds2 = ds.map(lambda x: {"width": x["photo"]["width"], "height": x["photo"]["height"]}, remove_columns=ds.column_names)
>>> ds3 = ds2.map(lambda x: {"pixels": x["width"] * x["height"]})
>>> max(ds3["pixels"])
178962432